lib/rspec/matchers/built_in/base_matcher.rb

```diff
@@ -112,6 +115,14 @@ def matcher_name
           end
         end
 
+        # @private
+        def reinitialize
```

Comment:
It is possible to push this method below `private` waterline.

I'd suggest naming it `reset`.